faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -257,7 +335,7 @@ private Connection(URL faunaRoot, String authToken, HttpClient client, MetricReg
    * @return a new {@link Connection}
    */
   public Connection newSessionConnection(String authToken) {
-    return new Connection(faunaRoot, authToken, client, registry, jvmDriver, getLastTxnTime(), defaultQueryTimeout, userAgent);
+    return new Connection(faunaRoot, authToken, client, registry, jvmDriver, scalaVersion, getLastTxnTime(), defaultQueryTimeout, userAgent);
```

Comment:
Could we also pass the `runtimeEnvironmentHeader` in to the private constructor to avoid rebuilding it?

Reply:
I agree with the suggestion here :+1: 
Good to approve the PR afterwards.

Reply:
@erickpintor & @fireridlle I have pushed a commit to tackle this comment https://github.com/fauna/faunadb-jvm/pull/300/commits/447523a1896f19b8b81ef0aff369ffda0f08a920

I had to make the `RuntimeEnvironmentHeader` static and I pass only the `String` representation to the constructor.

WDYT?

Reply:
I'm fine with it.